### PR TITLE
[RFC]vim-patch:8.1.{495,505,531,533,583,623,630,641,686,715,833,1012,1221,1651}

### PR DESCRIPTION
**Pull request by janlazo**
It's been RDY for a while. I'll just continue pushing trivial or independent vim patches in this PR until this gets merged or Vim releases 8.2.0000.

~~Going to stop for now. I can't trust the CI since https://github.com/neovim/neovim/commit/47d53d66bdcf1df1688c9af6999156a6c97f5aa7 .~~ Resolved quickly. Thanks.

### CODE REVIEW

**Comment by janlazo** [comment]
`TermClose` is flaky now?
```
[  ERROR   ] 1 error, listed below:
[  ERROR   ] ...im/neovim/test/functional/autocmd/termclose_spec.lua @ 20: TermClose event triggers when fast-exiting terminal job stops
test/functional/helpers.lua:96: Vim(call):E900: Invalid channel id
stack traceback:
	test/functional/helpers.lua:96: in function 'request'
	test/functional/helpers.lua:254: in function 'command'
	...im/neovim/test/functional/autocmd/termclose_spec.lua:23: in function <...im/neovim/test/functional/autocmd/termclose_spec.lua:20>
 17 SKIPPED TESTS
 1 ERROR
-- Output to stderr:
CMake Error at /home/travis/build/neovim/neovim/cmake/RunTests.cmake:63 (message):
  functional tests failed with error: 1
```

Am I missing a `cmdline` patch?
```
Direct leak of 9 byte(s) in 1 object(s) allocated from:
    #0 0x50ae38 in malloc (/home/travis/build/neovim/neovim/build/bin/nvim+0x50ae38)
    #1 0x1134874 in try_malloc /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:70:15
    #2 0x1134a34 in xmalloc /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:104:15
    #3 0x144ec95 in home_replace_save /home/travis/build/neovim/neovim/build/../src/nvim/os/env.c:953:17
    #4 0x756d8f in buflist_nr2name /home/travis/build/neovim/neovim/build/../src/nvim/buffer.c:2396:10
    #5 0x108a331 in fm_getname /home/travis/build/neovim/neovim/build/../src/nvim/mark.c:587:10
    #6 0x1090b11 in ex_jumps /home/travis/build/neovim/neovim/build/../src/nvim/mark.c:791:14
    #7 0xc92c3e in do_one_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:2248:5
    #8 0xc70ee0 in do_cmdline /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:593:20
    #9 0xc771f5 in do_cmdline_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:270:10
    #10 0x9ed571 in f_execute /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:8508:5
    #11 0x906917 in call_func /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6562:11
    #12 0x91a42e in get_func_tv /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6303:11
    #13 0x9afdf1 in eval7 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4406:15
    #14 0x9ac393 in eval6 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4103:7
    #15 0x9a9b07 in eval5 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3951:7
    #16 0x9a4e0f in eval4 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3687:7
    #17 0x9a42ec in eval3 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3605:7
    #18 0x9a37bc in eval2 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3536:7
    #19 0x8fecb2 in eval1 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3463:7
    #20 0x919d1d in get_func_tv /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6276:9
    #21 0x9afdf1 in eval7 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4406:15
    #22 0x9ac393 in eval6 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4103:7
    #23 0x9a9b07 in eval5 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3951:7
    #24 0x9a4e0f in eval4 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3687:7
    #25 0x9a42ec in eval3 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3605:7
    #26 0x9a37bc in eval2 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3536:7
    #27 0x8fecb2 in eval1 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3463:7
    #28 0x8fe0ca in eval0 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3423:9
    #29 0x90b022 in ex_let_const /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:1604:9
```

**Comment by erw7** [comment]
> Am I missing a cmdline patch?

@janlazo You need https://github.com/vim/vim/commit/d93090f41f70c521cfad5b25efcb0024b9480082.

File: src/nvim/testdir/shared.vim [hunk 1]
```diff
@@ -262,6 +266,12 @@ func! Screenline(lnum)
...
-  return has('gui') && ($DISPLAY != "" || has('gui_running'))
+" Gets the text of a terminal line, using term_scrape()
+func Get_terminal_text(bufnr, row)
+  let list = term_scrape(a:bufnr, a:row)
+  let text = ''
+  for item in list
+    let text .= item.chars
+  endfor
+  return text
 endfunc
```
Comment (blueyed) [inline src/nvim/testdir/shared.vim]: We should not add `Get_terminal_text` (unused, and removed in Vim via "patch 8.1.0769: :stop is covered in two tests" later).  Merged already via https://github.com/neovim/neovim/pull/10157.

Comment (janlazo) [reply src/nvim/testdir/shared.vim]: It's harmless and patch 8.1.1581 will remove it.

**Comment by justinmk** [comment]
reopen?

**Comment by justinmk** [comment]
Thanks @janlazo !

**Comment by janlazo** [comment]
I forgot to submit the patch for 8.1.0715 :( . I'll followup in another PR.

**Pull request merged**